packages/connect/src/events/popup.ts

```diff
@@ -53,19 +55,31 @@ export interface PopupError {
     };
 }
 
+export interface PopupShowError {
+    type: typeof POPUP.SHOW_ERROR;
+    payload: {
+        error: string;
+    };
+}
+
 export interface PopupClosedMessage {
     type: typeof POPUP.CLOSED;
     payload: { error: any } | null;
 }
 
 export type PopupEvent =
     | {
-          type: typeof POPUP.LOADED | typeof POPUP.CANCEL_POPUP_REQUEST;
+          type: typeof POPUP.LOADED;
           payload?: typeof undefined;
       }
+    | {
+          type: typeof POPUP.CANCEL_POPUP_REQUEST;
+          payload: { success: boolean };
```

Comment:
So I passed it here `success` in the payload because it is the less changes it requires and it works ok. 
It could be also passed as a new parameter and it will work as well. I do not have any strong opinions on this.